File: Packages/cgm/Denoising.py
#!/usr/bin/evn python3

class Denoising:
	def __init__(self, filepath):
		from pandas import read_table
		self.filepath = filepath
		self.stats = read_table(self.filepath, index_col = 0, skiprows = [1])
	def plot(self):
		from matplotlib.pyplot import plot, yscale, ylabel, subplots_adjust, savefig, show
		for i in range(self.stats.shape[0]):
			plot(self.stats.iloc[i])
		yscale("log")
		ylabel("log(#Sequences)")
		subplots_adjust(left = .15)
		savefig("./Downloads/Researches/CGM/Figures/denoising.png")
		show()

File: Packages/cgm/Denoising.R
library(R6)
library(reticulate)

Denoising <- R6Class(
	classname = "Denoising",
	public = list(
		stats = NULL,
		initialize = function(filepath) {
			self$stats = read.table(filepath, header = TRUE, sep = "\t", row.names = 1)
			use_python(system("which python3"))
		},
		plot = function() {
			pyplot <- import("matplotlib.pyplot")
			for (i in 1:dim(self$stats)[1]) {
				pyplot$plot(unlist(self$stats[i,]))
			}
			pyplot$yscale("log")
			pyplot$ylabel("log(#Sequences)")
			pyplot$savefig("./Downloads/Researches/CGM/Figures/denoising.png")
			pyplot$show()
		}
	)
)
